File: codeabbey/002_sums_in_loop.rb
#!/usr/bin/env ruby
# encoding: utf-8

x = ARGF.file.read.split("\n").map do |line|
  nums = line.split(' ')
  if nums.size == 2
    nums[0].to_i + nums[1].to_i
  else
    nil
  end
end.compact.join ' '
puts x

File: learning/javascript/codeabbey/002-sums-in-loop.js
#! /usr/bin/env node

var fs, input, output;

fs = require('fs');
input = fs.readFileSync('./input.txt', 'utf8')
          .split('\n')
          .slice(1, -1)
          .map(function(e) { return e.split(' '); });

output = input.map(function(nums) {
  return nums.reduce(
      function(acc, e) { return acc + parseInt(e, 10); },
      0);
}).join(' ');

console.log(output);
